iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -737,6 +746,22 @@ public void executeOperation(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
     }
   }
 
+  private static void expireSnapshotByIds(Table icebergTable, String[] idsToExpire,
```

Comment:
could we extract the expire snapshots logic into `expireSnapshots(icebergTable, expireSnapshotsSpec, deleteExecutorService)`

Reply:
refactored the code as suggested